### DISCUSSION TITLE
Dynamically add and remove Services during Runtime

### DISCUSSION THREAD

**Discussion by nhelou** [COVESA/vsomeip#594] [Q&A]
As mentioned in the last COVESA VSOMIP monthly call I am wondering if there is a way to add and remove services during runtime without restarting the whole system. Currently you either need to add a new service to the global or an application specific configuration file. Both need to be known during system startup.

I was thinking about implementing a custom vsomeip3-cfg.so to load the data dynamically, but I am not sure if this is a good idea. A better solution would be an officially supported feature.

What do you think?

**Comment by zhaoxin39913** [comment]
Yes, this can be done by calling update_service_configuration() in runtime phase. Then you don't need to declare the service in configuration file. Please refer to my code in taf_SomeipSvr::VSOMEIPOfferService() in https://git.codelinaro.org/clo/le/platform/TelAF/-/blob/telaf.lnx.1.1/components/tafSomeipGWSvc/tafSomeipSvrImpl.cpp
